src/main/java/be/seeseemelk/mockbukkit/WorldMock.java

```diff
@@ -826,18 +824,27 @@ public void save()
 	}
 
 	@Override
-	public <T extends Entity> T spawn(Location location, Class<T> clazz) throws IllegalArgumentException
+	public <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz) throws IllegalArgumentException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Validate.notNull(location, "The provided location must not be null.");
+		Validate.notNull(location, "The provided clazz must not be null.");
+
+		EntityMock entity = mockEntity(clazz);
+		entity.setLocation(location);
+		server.registerEntity(entity);
+
+		return (T) entity;
 	}
 
 	@Override
-	public <T extends Entity> T spawn(Location location, Class<T> clazz, Consumer<T> function)
+	public <T extends Entity> T spawn(@NotNull Location location, @NotNull Class<T> clazz, Consumer<T> function)
```

Comment:
The `Consumer<T>` should be marked as nullable here

Reply:
fix [`95beb27` (#269)](https://github.com/MockBukkit/MockBukkit/pull/269/commits/95beb27604d3578102eb7098565c7029b300f999)